cmd/run.go

```diff
@@ -300,9 +319,21 @@ func (c *cmdRun) run(cmd *cobra.Command, args []string) (err error) {
 		}()
 	}
 
+	// TODO: Make the timeout configurable?
+	if werr := waitInitDone(10 * time.Second); werr != nil {
+		logger.WithError(werr).Warn()
+	}
+	test.preInitState.Events.Emit(&event.Event{Type: event.TestStart})
 	// Start the test! However, we won't immediately return if there was an
 	// error, we still have things to do.
 	err = execScheduler.Run(globalCtx, runCtx, samples)
+	waitTestEndDone := test.preInitState.Events.Emit(&event.Event{Type: event.TestEnd})
```

Comment:
Possibly, but this error would also be sent in the `Exit` event, so it could be duplicated. I think the final error we return from `run()` is more valuable to external modules, since that also includes the `AbortReason` and `ExitCode`. So instead of subscribers checking both errors and handling these cases, they can only check the `Exit` one instead.

`Exit` is also guaranteed to be sent, since it's done in the `defer`, whereas this event could be skipped if an error happens before and we don't get to actually run the test.